inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -3381,28 +3441,18 @@ private <T> Optional<BeanDefinition<T>> pickOneBean(
         return Optional.ofNullable(definition);
     }
 
+    @SuppressWarnings("java:S1871")
     private <T> void filterProxiedTypes(
-        Collection<BeanDefinition<T>> candidates,
-        boolean filterDelegates) {
+        Collection<BeanDefinition<T>> candidates) {
         int count = candidates.size();
         Set<Class<?>> proxiedTypes = new HashSet<>(count);
```

Comment:
`new HashSet<>(count)` Looks incorrect as it will be resized before it will reach the count capacity. There are some options on how to do it in Java 19 https://bugs.openjdk.org/browse/JDK-8186958 Maybe we can have similar internal util methods.

Reply:
I didn't change this, however I have another PR coming after this one that removes this logic